docs/guides/connecting_to_your_data/datasource_configuration/how_to_configure_a_pandas_datasource.md

```diff
@@ -0,0 +1,355 @@
+
+### 2. Create a new Datasource configuration.
+
+A new Datasource can be configured in Python as a dictionary with a specific set of keys.  We will build our Datasource configuration from scratch in this guide, although you can just as easily modify an existing one.
+
+To start, create an empty dictionary.  You will be populating it with keys as you go forward.
+
+At this point, the configuration for your Datasource is merely:
+
+```python
+datasource_config = {}
+```
+
+However, from this humble beginning you will be able to build a full Datasource configuration.
+
+#### The keys needed for your Datasource configuration
+
+At the top level, your Datasource's configuration will need the following keys: 
+- `name`: The name of the Datasource, which will be used to reference the datasource in Batch Requests.
+- `class_name`: The name of the Python class instantiated by the Datasource.  Typically, this will be the `Datasource` class.
+- `module_name`: the name of the module that contains the Class definition indicated by `class_name`.
+- `execution_engine`: a dictionary containing the `class_name` and `module_name` of the Execution Engine instantiated by the Datasource.
+- `data_connectors`: the configurations for any Data Connectors and their associated Data Assets that you want to have available when utilizing the Datasource.
+
+In the following step we will add those keys and their corresponding values to your currently empty Datasource configuration dictionary.
+
+### 3. Name your Datasource
+
+The first key that you will need to define for your new Datasource is its `name`.  You will use this to reference the Datasource in future workflows.  It can be anything you want it to be, but ideally you will name it something relevant to the data that it interacts with.
+
+For the purposes of this example, we will name this Datasource:
+
+```python
+    "name": "name_of_my_datasource",
+```
+
+You should, however, name your Datsource something more relevant to your data.
+
+At this point, your configuration should now look like:
+
+```python
+datasource_config = {
+    "name": "name_of_my_datasource",  # Preferably name it something relevant
+}
+```
+
+### 4. Specify the Datasource class and module
+
+The `class_name` and `module_name` for your Datasource will almost always indicate the `Datasource` class found at `great_expectations.datasource`.  You may replace this with a specialized subclass, or a custom class, but for almost all regular purposes these two default values will suffice.  For the purposes of this guide, add those two values to their corresponding keys.
+
+```python
+    "class_name": "Datasource",
+    "module_name": "great_expectations.datasource",
+```
+
+Your full configuration should now look like:
+
+```python
+datasource_config = {
+    "name": "my_datasource_name",  # Preferably name it something relevant
```

Comment:
I think the name may have changed between here and the previous steps.

Reply:
It did.  I've updated the earlier code snippets, though they'll be updated again when once I move the code snippets into a script under test after the drafts are done.